File: Npc/11004186.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004186: Darphony
/// </summary>
public class _11004186 : NpcScript {
    protected override int First() {
        return 10;
    }

    // Select 0:
    // $script:0806025707010634$
    // - Ah... N-nice to meet you... 
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (10, 0):
                // $script:0806025707010635$
                // - I came here to get t-t-tougher, so I can chase off the w-wolves plaguing our farm.
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (10, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11004186.py
""" 11004186: Darphony """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 10

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0806025707010634$
        # - Ah... N-nice to meet you... 
        return None # TODO

    def __10(self, pick: int) -> int:
        # $script:0806025707010635$
        # - I came here to get t-t-tougher, so I can chase off the w-wolves plaguing our farm.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (10, 0):
            return Option.CLOSE
        return Option.NONE
